packages/utils/src/Multikey.ts

```diff
@@ -0,0 +1,172 @@
+    default:
+      throw new DidError(`The provided key type "${type}" is not supported.`)
+  }
+}
+
+// TODO: This could also be exposed in a new release candidate of the `@kiltprotocol/jcs-data-integrity-proofs-common` package.
+function multibase58BtcKeyBytesEncoding(
+  key: Uint8Array,
+  keyPrefix: number
+): Base58BtcMultibaseString {
+  const varintEncodedPrefix = varint.encode(keyPrefix)
+  const prefixedKey = u8aConcat(varintEncodedPrefix, key)
+  const base58BtcEncodedKey = base58Encode(prefixedKey)
+  return `z${base58BtcEncodedKey}`
+}
+
+export function encodeMultibaseKeypair(
+  args: Pick<KeyringPair, 'publicKey'> & {
+    type: KnownTypeString
+    secretKey: Uint8Array
+  }
+): MultibaseKeyPair
+export function encodeMultibaseKeypair(
+  args: Pick<KeyringPair, 'publicKey'> & {
+    type: KnownTypeString
+  }
+): MultibasePublicKey
+/**
+ * Calculate the Multikey representation of a keypair given its type and public/secret keys.
+ *
+ * @param keypair The input keypair to encode as Multikey.
+ * @param keypair.type The keypair type indicated by a type string.
+ * @param keypair.publicKey The keypair public key.
+ * @param keypair.secretKey Optionally, the keypair's secret key.
+ * @returns The Multikey representation (i.e., multicodec-prefixed, then multibase encoded) of the provided keypair.
+ */
+export function encodeMultibaseKeypair({
+  type,
+  publicKey,
+  secretKey,
+}: Pick<KeyringPair, 'publicKey'> & {
+  type: KnownTypeString
+  secretKey?: Uint8Array
+}): MultibasePublicKey & Partial<MultibaseSecretKey> {
+  const [multiCodecPublicKeyPrefix, multiCodedSecretKeyPrefix] =
+    mapTypeStringToPrefixes(type)
+
+  const keypair: MultibasePublicKey & Partial<MultibaseSecretKey> = {
+    publicKeyMultibase: multibase58BtcKeyBytesEncoding(
+      publicKey,
+      multiCodecPublicKeyPrefix
+    ),
+  }
+  if (secretKey) {
+    keypair.secretKeyMultibase = multibase58BtcKeyBytesEncoding(
+      secretKey,
+      multiCodedSecretKeyPrefix
+    )
+  }
+
+  return keypair
+}
+
+const publicKeyPrefixes: Record<number, KeyTypeString> = {
+  [MULTICODEC_SECP256K1_PREFIXES[0]]: 'secp256k1',
+  [MULTICODEC_X25519_PREFIXES[0]]: 'x25519',
+  [MULTICODEC_ED25519_PREFIXES[0]]: 'ed25519',
+  [MULTICODEC_SR25519_PREFIXES[0]]: 'sr25519',
+}
+
+const secretKeyPrefixes: Record<number, KeyTypeString> = {
+  [MULTICODEC_SECP256K1_PREFIXES[1]]: 'secp256k1',
+  [MULTICODEC_X25519_PREFIXES[1]]: 'x25519',
+  [MULTICODEC_ED25519_PREFIXES[1]]: 'ed25519',
+  [MULTICODEC_SR25519_PREFIXES[1]]: 'sr25519',
+}
+
+export function decodeMultibaseKeypair(keypair: MultibaseKeyPair): Pick<
+  KeyringPair,
+  'publicKey'
+> & {
+  secretKey: Uint8Array
+  type: KeyTypeString
+}
+export function decodeMultibaseKeypair(
+  keyPairPublicKey: MultibasePublicKey
+): Pick<KeyringPair, 'publicKey'> & { type: KeyTypeString }
+/**
+ * Decode a Multikey representation of a verification method into its fundamental components: the public key and the key type.
```

Comment:
we probably should add that only base 58 is supported? same for encode.

Reply:
d69481d40bf17e19c54429c89df106f983c9919b ; I have then moved docstrings to the overloads(1bdf0d8ca39040dfe041f7839520cdaa52d46c76), tailoring them to the respective parameters set